src/stores/index.js

```diff
@@ -929,40 +834,120 @@ const Store = Reflux.createStore({
    *
    * @param {Object} connection - The current connection.
    */
-  _connect(connection) {
+  async _connect(connection) {
+    // Set the connection's app name to the electron app name of Compass.
+    connection.appname = electron.remote.app.getName();
+
     this.dataService = new DataService(connection);
     this.appRegistry.emit('data-service-initialized', this.dataService);
-    this.dataService.connect((error, ds) => {
-      if (error) {
-        this.StatusActions.done();
-        this.setState({
-          isValid: false,
-          errorMessage: error.message,
-          syntaxErrorMessage: null
-        });
-      } else {
-        const currentConnection = this.state.currentConnection;
-        const currentSaved = this.state.connections[currentConnection._id];
 
-        this.state.isValid = true;
-        this.state.isConnected = true;
-        this.state.errorMessage = null;
-        this.state.syntaxErrorMessage = null;
-        this.state.hasUnsavedChanges = false;
-        this.state.isURIEditable = false;
-        this.state.customUrl = this.state.currentConnection.driverUrl;
+    try {
+      const runConnect = promisify(this.dataService.connect.bind(this.dataService));
+      const connectedDataService = await runConnect();
 
-        currentConnection.lastUsed = new Date();
+      const currentConnection = this.state.currentConnection;
+      const currentSaved = this.state.connections[currentConnection._id];
 
-        if (currentSaved) {
-          this._saveConnection(currentConnection);
-        } else {
-          this._saveRecent(currentConnection);
-        }
+      this.state.isValid = true;
+      this.state.isConnected = true;
+      this.state.errorMessage = null;
+      this.state.syntaxErrorMessage = null;
+      this.state.hasUnsavedChanges = false;
+      this.state.isURIEditable = false;
+      this.state.customUrl = this.state.currentConnection.driverUrl;
+
+      currentConnection.lastUsed = new Date();
```

Comment:
this part and the following if is really hard to put in context. have no clue what is supposed to do, i think putting this in a method with a good name and comment would help a ton. What save means here, what is the difference between `saveConnection` and `saveRecent`?

Reply:
Yeah probably naming could be better :) `_saveRecent` calls `_saveConnection` later in code. `_saveConnection` saves connection to reflux store.